WalletWasabi.Fluent/ViewModels/Wallets/CoinJoinStateViewModel.cs

```diff
@@ -0,0 +1,282 @@
+using System.Reactive;
+using System.Reactive.Linq;
+using System.Windows.Input;
+using Avalonia.Threading;
+using NBitcoin;
+using ReactiveUI;
+using WalletWasabi.Fluent.Helpers;
+using WalletWasabi.Fluent.State;
+using WalletWasabi.WabiSabi.Client;
+using WalletWasabi.Wallets;
+
+namespace WalletWasabi.Fluent.ViewModels.Wallets;
+
+public partial class CoinJoinStateViewModel : ViewModelBase
+{
+	enum State
+	{
+		AutoCoinJoin,
+		ManualCoinJoin,
+
+		AutoStarting,
+		Paused,
+		AutoPlaying,
+
+		Stopped,
+		ManualPlaying,
+	}
+
+	enum Trigger
+	{
+		AutoCoinJoinOn,
+		AutoCoinJoinOff,
+		AutoCoinJoinEntered,
+		ManualCoinJoinEntered,
+		Pause,
+		Play,
+		Stop,
+		PlebStop,
+		RoundStartFailed,
+		RoundStart
+	}
+
+	private readonly StateMachine<State, Trigger> _machine;
```

Comment:
Just wondering if renaming this to `_state` or `_stateMachine` would increase readability

Reply:
done.